hopsworks-common/src/test/io/hops/hopsworks/common/featurestore/query/TestConstructorController.java

```diff
@@ -0,0 +1,454 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2019, Logical Clocks AB. All rights reserved
+ *
+ * Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.hops.hopsworks.common.featurestore.query;
+
+import io.hops.hopsworks.common.dao.featurestore.Featurestore;
+import io.hops.hopsworks.common.dao.featurestore.featuregroup.Featuregroup;
+import io.hops.hopsworks.common.featurestore.FeaturestoreFacade;
+import io.hops.hopsworks.common.featurestore.feature.FeatureDTO;
+import io.hops.hopsworks.common.featurestore.featuregroup.FeaturegroupController;
+import io.hops.hopsworks.common.featurestore.featuregroup.FeaturegroupDTO;
+import io.hops.hopsworks.common.featurestore.featuregroup.FeaturegroupFacade;
+import org.apache.calcite.sql.JoinType;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mockito;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class TestConstructorController {
```

Comment:
@SirOibaf shall we move this into hopsworks-UT?

Reply:
Unit tests should be in the same module of the code they are testing. AFAIK `Hopsworks-UT`, despite the name, contains additional endpoints for the integration testing, not unit testing. This test class does not require a running instance of Hopsworks.

Reply:
> Unit tests should be in the same module of the code they are testing
That is correct.

>  AFAIK Hopsworks-UT, despite the name, contains additional endpoints for the integration testing, not unit testing.
I'm missing something here.

Reply:
My bad - I thought that `hopsworks-UT` contained the additional endpoints Alex wrote for the integration testings, but they are actually in the `hopsworks-testing` module . It seems that it contains the `TestRestCodes` class instead. Which is a unit test and IMHO should stay with it's package. I don't really see the purpose of having it in a separate module.